packages/openworkflow/schema.ts

```diff
@@ -0,0 +1,140 @@
+/**
+ * Structural types for supported workflow input schemas. These mirror the
+ * public shapes from common validation libraries so we can infer types without
+ * depending on those packages directly.
+ *
+ * This file has been inspired by the `schemaTask` provided by trigger.dev.
+ * See reference file: https://github.com/triggerdotdev/trigger.dev/blob/main/packages/core/src/v3/types/schemas.ts
+ */
+
+/* eslint-disable @typescript-eslint/no-explicit-any, @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-return, @typescript-eslint/no-unsafe-call */
+
+export interface WorkflowSchemaZodLike<TInput, TParsed> {
+  _input: TInput;
+  _output: TParsed;
+  parse?: (value: unknown) => TParsed;
+  parseAsync?: (value: unknown) => Promise<TParsed>;
+  safeParse?: (value: unknown) => { success: boolean; data?: TParsed };
+}
+
+export interface WorkflowSchemaValibotLike<TInput, TParsed> {
+  schema: {
+    _types?: {
+      input: TInput;
+      output: TParsed;
+    };
+  };
+}
+
+export interface WorkflowSchemaArkTypeLike<TInput, TParsed> {
+  inferIn: TInput;
+  infer: TParsed;
+  assert?: (value: unknown) => asserts value is TParsed;
+}
+
+export type WorkflowSchemaPlainValidator<TInput> = (
+  value: unknown,
+) => Promise<TInput> | TInput;
+
+export interface WorkflowSchemaSimpleParse<TInput> {
+  parse: (value: unknown) => TInput;
+}
+
+export interface WorkflowSchemaSuperstructLike<TInput> {
+  create: (value: unknown) => TInput;
+}
+
+export interface WorkflowSchemaYupLike<TInput> {
+  validateSync: (value: unknown) => TInput;
+}
+
+export interface WorkflowSchemaScaleLike<TInput> {
+  assert(value: unknown): asserts value is TInput;
+}
+
+export type WorkflowSchemaWithInOut<TInput, TParsed> =
```

Comment:
Open question: I'm wondering if we really need to support all of these parsers vs. simply supporting `zod` only.

Generally, I'd prefer a smaller surface area until people specifically ask for something, since more code = more to maintain, and Zod seems like the most popular for us to standardize on. Strong defaults and all that...

Not 100% sure either way, though. What do you think?

Reply:
I guess... it depends. I'm ok with whatever you choose.
I don't think `parse` or any other equivalent function from all this libs received many breaking changes (e,g zod uses the same parse/safeParse/parseAsync for as long I used them, the only changed things like from `z.string().uuid()` to `z.uuid()`, etc...).

I initially though it's not a very big effort to mirror those functions, since they are not that many, but we can simplify it, your call.
My use case is with zod so for me it's irrelevant at the moment.